packages/typespec-azure-resource-manager/test/resource.test.ts

```diff
@@ -1204,3 +1204,32 @@ it("recognizes resource with customResource identifier", async () => {
 `);
   expectDiagnosticEmpty(diagnostics);
 });
+
+describe("multiple services", () => {
+  it("assign resources to the correct service", async () => {
+    const { program } = await Tester.compile(`
+    @armProviderNamespace
+    namespace Microsoft.ServiceA {
+      model ResA is TrackedResource<{}> {
+        @key @segment("foos") @path name: string;
+      }
+    }
+
+    @armProviderNamespace
+    namespace Microsoft.ServiceB {
+      model ResB is TrackedResource<{}> {
+        @key @segment("foos") @path name: string;
+      }
+    }
+  `);
```

Comment:
so how would it behave if we have two parts with the same provider namespace but with different api version enum? this case happen in compute.
Do we have to do something like this?
```
@providerNamespace("Microsoft.Compute")
@versioned(VMVersions)
namespace Microsoft.Compute.Vms {}

@providerNamespace("Microsoft.Compute")
@versioned(DiskVersions)
namespace Microsoft.Compute.Disks {}
```
and if we have to do it in the above way, we might not be able to just put them into the one client.tsp without changing their specs.